JASP-Engine/JASP/R/MixedModelsCommon.R

```diff
@@ -491,18 +492,27 @@
       if (any(attr(model, "class") %in% c("std::runtime_error", "C++Error", "error"))) {
         if (model$message == "(maxstephalfit) PIRLS step-halvings failed to reduce deviance in pwrssUpdate") {
           ANOVAsummary$setError(
-            gettext("The optimizer failed to find solution. Probabably due to quasi-separation in the data. Try removing some of the predictors.")
+            gettext("The optimizer failed to find a solution. Probably due to quasi-separation in the data. Try removing some of the predictors.")
           )
         } else if (model$message == "cannot find valid starting values: please specify some") {
           # currently no solution to this, it seems to be a problem with synthetic data only.
           # I will try silving it once someone actually has problem with real data.
-          ANOVAsummary$setError(gettext("Invalid starting values."))
+          ANOVAsummary$setError(gettext("The optimizer failed to find a solution due to invalid starting values. (JASP currently does not support specifying different starting values.)"))
+        } else if (model$message == "Downdated VtV is not positive definite"){
+          ANOVAsummary$setError(gettext("The optimizer failed to find a solution. Probably due to scaling issues quasi-separation in the data. Try rescaling or removing some of the predictors."))
         } else{
-          ANOVAsummary$setError(model$message)
+          ANOVAsummary$setError(.unv(model$message))
         }
         
-        jaspResults[["ANOVAsummary"]] <- ANOVAsummary
-        jaspResults[["mmModel"]]      <- NULL
+        
+        hasERROR <- createJaspState()
+        hasERROR$object <- TRUE
+        hasERROR$dependOn(c(dependencies, seed_dependencies))
+        jaspResults[["hasERROR"]] <- hasERROR
```

Comment:
It's a little akward but I cannot set the `jaspResults[["mmModel"]]` to `NULL` in order to skip the block with summary and plotting functions because the model would be reffited, and it would skip the main summary that was already produced (inside of which it's set to NULL, I cannot do that before since I need to keep the error message). This object keeps track on whether the model failed and allows to skip the summary and plotting functions. I could alternatively create a designated function that would set the model to null after creating the summary table and it wouldn't be dependent on it. It might be a better solution.

Reply:
however, it would resulted on refiting the failed model every time, even when requested

Reply:
> This object keeps track on whether the model failed

Okay, but if you set an error on any jaspResults object you can check for an error using `$getError()`.

Reply:
oh, great. That's much better solution. Thanks